public/pages/Snapshots/components/RenameInput/RenameInput.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiFormRow, EuiFieldText, EuiSpacer } from "@elastic/eui";
+import React, { useState, ChangeEvent } from "react";
+import CustomLabel from "../../../../components/CustomLabel";
+
+interface RenameInputProps {
+  getRenamePattern: (prefix: string) => void;
+  getRenameReplacement: (prefix: string) => void;
+}
+
+const RenameInput = ({ getRenamePattern, getRenameReplacement }: RenameInputProps) => {
+  const [renamePattern, setRenamePattern] = useState("");
+  const [renameReplacement, setRenameReplacement] = useState("");
+
+  const onPatternChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setRenamePattern(e.target.value);
+    getRenamePattern(e.target.value);
+  };
+
+  const onReplacementChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setRenameReplacement(e.target.value);
+    getRenameReplacement(e.target.value);
+  };
+
+  const patternHelpText =
+    "Use regular expressiojn to define how index names will be renamed. By default, input (.+) to reuse the entire index name [Learn more]";
```

Comment:
Minor detail, but I believe the convention is for `helpText` and error messages to end with a hard stop (`.`).

Reply:
fixed and pushed.